x-pack/platform/plugins/shared/inference/server/chat_complete/anonymization/regex_worker.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Piscina from 'piscina';
+import { RegexAnonymizationTask } from '@kbn/inference-common';
+import { regexWorkerFilename } from './regex_worker_task';
+
+let worker: Piscina | undefined;
+
+export function initRegexWorker() {
```

Comment:
this perhaps should be part of the inference service. that would also allow us to move to a model where we can manage worker threads centrally (kibana-wide). what happens if someone calls this function twice?

Reply:
For now I added a return if the worker already exists